### PR TITLE
Update design-doc.md with the new proposal process

### PR DESCRIPTION
**Pull request by bwplotka**
See the accepted proposal https://github.com/prometheus/proposals/blob/main/proposals/2022-11-23_proposal-process.md

### CODE REVIEW

**Comment by bwplotka** [comment]
NOTE: Migration of docs mentioned in index is in progress. All tracked here: https://github.com/prometheus/proposals/issues

File: content/docs/introduction/design-doc.md [hunk 1]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+It's essential to clearly explain the reasons behind certain design decisions to have a community consensus. This is especially
+important in Prometheus, where every decision might have a significant impact given the high adoption and stability of the software and standards we work on.
+
+In our world, no decision is perfect, so having a design document that explains the trade-offs we made is essential.
+Such a document can also be used later on as a reference and for knowledge-sharing purposes.
+
+Note that design documents do not always reflect exactly what has been (or will be) implemented. Implementation details
+might have changed since a feature was merged. Design docs are not considered documentation and can not define a standard.
+Instead, it should explain the motivation, scope, the decision made and alternatives considered.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:16]: decision → decisions

Comment (beorn7) [reply content/docs/introduction/design-doc.md:16]: And maybe you should use the Oxford comma? I was under the impression we generally prefer it in the Prometheus universe (and yeah, we should probably have a formalized documentation style guide one day).

File: content/docs/introduction/design-doc.md [hunk 2]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+## Current proposals
+
+The [github.com/prometheus/proposals](https://github.com/prometheus/proposals)repository holds all the past and current proposals for the Prometheus Ecosystem. It's the single place for reviewing, discovering and working on the design documents. It also a record of the past decisions and approvals.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:20]: Blank missing between `/proposals)` and `repository`.

Comment (beorn7) [reply content/docs/introduction/design-doc.md:20]: It also→ It is also

Comment (beorn7) [reply content/docs/introduction/design-doc.md:20]: And another potential Oxford comma usage after "discovering".

File: content/docs/introduction/design-doc.md [hunk 3]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+* [The PRs with proposal label](https://github.com/prometheus/proposals/pulls?q=is%3Aopen+is%3Apr+label%3Aproposal) shows all the pending proposals.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:22]: [The PRs with proposal label] → The [PRs with the _proposal_ label]

(One more "the" and consistently excluding the initial "The" from the linked text. Also, maybe, use italics to make clearer that "proposal" is literally the text of the label.)

shows → show

File: content/docs/introduction/design-doc.md [hunk 4]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+* The [proposals directory](https://github.com/prometheus/proposals/tree/main/proposals) shows all the accepted proposals. See the "Implementation Status" for details on the implementation.
+* [The unmerged PRs with proposal label](https://github.com/prometheus/proposals/pulls?q=is%3Apr+label%3Aproposal+is%3Aclosed+is%3Aunmerged) shows all the rejected proposals.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:24]: [The unmerged PRs with proposal label] → The [PRs with the _proposal_ label that are closed without merging]

shows → show

File: content/docs/introduction/design-doc.md [hunk 5]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+## Proposal process
+
+Don't get scared to propose ideas! It's amazing to innovate in the open and get feedback on the ideas.
+
+The process of proposing change with the design document would look as follows:
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:30]: Maybe "The process of proposing a change via a design document is the following:" ?

File: content/docs/introduction/design-doc.md [hunk 6]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+1. Fork `github.com/prometheus/proposals`.
+2. Create a GitHub Pull Request with a design document in the markdown format to the [proposals directory](./proposals). Make sure to use [template](0000-00-00_template.md) as the guide for what sections should be present in the document. Put the creation date (the day you started preparing this design document) as the prefix and some unique name as the suffix in the file name. Once the PR is proposed, maintainer will assign "proposal" label.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:33]: in the markdown format -> in markdown format

[template] → the [template]

assign "proposal" label → assign a “proposal” label

(Added "a" and typographically correct quotes.)

Or maybe better: assign a _proposal_ label.

File: content/docs/introduction/design-doc.md [hunk 7]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+   1. If you prefer Google Docs to any other collaboration tool, feel free to use it in the initial state. We recommend [Open Source Design document Template](https://docs.google.com/document/d/1zeElxolajNyGUB8J6aDXwxngHynh4iOuEzy3ylLc72U/edit#). However, the approval process will only happen officially in the Pull Request.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:34]: recommend [Open Source Design document Template] → recommend the [Open Source Design Document Template]

(Consistent capitalization and one added "the".)

File: content/docs/introduction/design-doc.md [hunk 8]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+Note that design documents do not always reflect exactly what has been (or will be) implemented. Implementation details
+might have changed since a feature was merged. Design docs are not considered documentation and can not define a standard.
+Instead, it should explain the motivation, scope, the decision made and alternatives considered.
+
+## Current proposals
+
+The [github.com/prometheus/proposals](https://github.com/prometheus/proposals)repository holds all the past and current proposals for the Prometheus Ecosystem. It's the single place for reviewing, discovering and working on the design documents. It also a record of the past decisions and approvals.
+
+* [The PRs with proposal label](https://github.com/prometheus/proposals/pulls?q=is%3Aopen+is%3Apr+label%3Aproposal) shows all the pending proposals.
+* The [proposals directory](https://github.com/prometheus/proposals/tree/main/proposals) shows all the accepted proposals. See the "Implementation Status" for details on the implementation.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:23]: "Implementation Status" → “Implementation Status”

(Typographically correct quotes.)

File: content/docs/introduction/design-doc.md [hunk 9]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+## Proposal process
+
+Don't get scared to propose ideas! It's amazing to innovate in the open and get feedback on the ideas.
+
+The process of proposing change with the design document would look as follows:
+
+1. Fork `github.com/prometheus/proposals`.
+2. Create a GitHub Pull Request with a design document in the markdown format to the [proposals directory](./proposals). Make sure to use [template](0000-00-00_template.md) as the guide for what sections should be present in the document. Put the creation date (the day you started preparing this design document) as the prefix and some unique name as the suffix in the file name. Once the PR is proposed, maintainer will assign "proposal" label.
+   1. If you prefer Google Docs to any other collaboration tool, feel free to use it in the initial state. We recommend [Open Source Design document Template](https://docs.google.com/document/d/1zeElxolajNyGUB8J6aDXwxngHynh4iOuEzy3ylLc72U/edit#). However, the approval process will only happen officially in the Pull Request.
+3. Automatic formatter is enabled in the repository. Use `make` locally to format it. Use `make check` to check all links (will be done on the CI too).
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:35]: Automatic formatter → An automatic formatter (or: Automatic formatting)

to format it → to trigger it

on the CI too→ by the CI pipeline, too

I guess formatting is also done by the CI pipeline, isn't it?
And maybe mention that the `make` targets assume you have a working Go environment.

File: content/docs/introduction/design-doc.md [hunk 10]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+4. The design is accepted if the PR is merged into this repository. It's ok to eventually decide to reject the proposal and close the PR with meaningful reasons for why it was rejected.
+   1. If more eyes are needed, or no consensus was made: Propose and announce your idea on
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:37]: on → on the

File: content/docs/introduction/design-doc.md [hunk 11]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+      [Prometheus DevSummit](https://docs.google.com/document/d/11LC3wJcVk00l8w5P3oLQ-m3Y37iom6INAMEu2ZAGIIE/edit) or the [developers mailing list](/community) to gather more information. You are welcome to start working on the design document before a bigger discussion--it is often easier to have a discussion with prior information provided. Be prepared that the idea might be rejected later--still, the record of the document in the Pull Request is useful even in rejected state to inform about past decisions and opportunities considered.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:38]: in rejected → in a rejected

`--` doesn't work in GH markdown. You could use a literal n-dash, i.e. "–", but in that case, I would suggest blanks around it. Or use an US-style m-dash, i.e. "—", in which case it's fine to not use blanks around it.

File: content/docs/introduction/design-doc.md [hunk 12]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+   2. To merge the PR, we need approval (consensus) from the maintainers of the related component(s).
+   3. Optionally: Find a sponsor among Prometheus maintainers to get momentum on a change.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:40]: among Prometheus maintainers → among the Prometheus maintainers

File: content/docs/introduction/design-doc.md [hunk 13]
```diff
@@ -3,51 +3,44 @@ title: Design Documents
...
+
+Once PR get merged, the design document can change, but it requires (less strict, but still) a PR with review and merge by a maintainer.
```
Comment (beorn7) [inline content/docs/introduction/design-doc.md:42]: Once PR get→ Once the PR gets

**Review by beorn7 — CHANGES_REQUESTED** [review]
Great text. But let's only merge it once (most of) the migrations are done.

**Comment by beorn7** [comment]
Now I'm realizing that parts of this text are also contained in https://github.com/prometheus/proposals/blob/main/README.md .

Maybe don't replicate the proposal process here but link to it (and still propagate my nitpickings from this PR to the README.md in the proposals repo ;).

**Comment by bwplotka** [comment]
Makes sense, will do. Sorry for lag (any my English 🙃 )

**Comment by beorn7** [comment]
Given that this has been sitting for a couple of months now, what's the plan to move forward?

File: content/docs/introduction/design-doc.md (hunk 8 above)
Comment (bwplotka) [reply content/docs/introduction/design-doc.md:23]: I guess you want curly quotes? 
image
 (:

**Comment by bwplotka** [comment]
Next step was to do the work (: 

This PR is now updated @beorn7 @roidelapluie 

The readme improvement based on your comments @beorn7 (massive thanks for those!): https://github.com/prometheus/proposals/pull/27

Docs are not yet moved, but they are all recorded in GH issues. I will be moving them (will write a script) at some point. I don't think we need to wait for that, but up to you.

**Review by beorn7 — APPROVED** [review]
LGTM, but I'm not sure about merging this before the design docs have been move to https://github.com/prometheus/proposals . If we merge this now, a lot of design docs will be hard to find for the time being.

**Comment by bwplotka** [comment]
I think they are recorded in issues, which should be enough. Our process is that rejected or TODO proposals are in the issues anyway, so voting for merging. (: 

Will fix readme and migrate some with this and following week.

**Pull request merged**